pkg/provider/gardener/ruleset/disak8sstig/rules/242442.go

```diff
@@ -96,7 +96,13 @@ func (*Rule242442) checkImages(cluster string, pods []corev1.Pod) []rule.CheckRe
 				}
 
 				imageRef := pod.Status.ContainerStatuses[containerStatusIdx].ImageID
-				imageBase := strings.Split(strings.Split(imageRef, ":")[0], "@")[0]
+				named, err := dockerref.ParseNormalizedNamed(imageRef)
+				if err != nil {
+					checkResults = append(checkResults, rule.ErroredCheckResult(err.Error(), rule.NewTarget("cluster", cluster, "pod", pod.Name, "container", container.Name, "imageRef", imageRef)))
```

Comment:
```suggestion
					checkResults = append(checkResults, rule.ErroredCheckResult(err.Error(), rule.NewTarget("imageRef", imageRef)))
```
Nit: No need to be too specific here as we should never enter this `if` clause. If we reach this code it is more likely an error in `ParseNormalizedNamed` and we would only care about the `imageRef` it caused it.